analyses/tp53_nf1_score/04-tp53-altered-annotation.Rmd

```diff
@@ -0,0 +1,416 @@
+  # select useful columns
+  select(sample_id,Kids_First_Biospecimen_ID_RNA,Kids_First_Biospecimen_ID_DNA,
+         tp53_score,cancer_predispositions,HGVSp_Short,copy_number,hotspot,activating) %>%
+  arrange(sample_id) %>%
+  unique() 
+
+tp53_alterations_score 
+
+```
+
+Convert the above to wide format
+
+```{r}
+
+tp53_alterations_score_wide <- tp53_alterations_score %>%
+  # group 
+  group_by(sample_id,
+           Kids_First_Biospecimen_ID_DNA,
+           Kids_First_Biospecimen_ID_RNA,
+           cancer_predispositions,
+           tp53_score) %>%
+  # 
+  # sample_id as rows add SNV counts,CNV loss counts 
+  # and hotspot/activating mutation annotation  
+  summarise(
+    # summarize length of SNV and CNv alterations 
+    SNV_indel_counts= length(unique(HGVSp_Short[!is.na(HGVSp_Short)])), 
+    CNV_loss_counts = length(unique(copy_number[!is.na(copy_number)])),
+    # summarize unique hotspot values per sample_id
+    hotspot = toString(unique(hotspot[!is.na(hotspot) ])),
+    activating = toString(unique(activating[!is.na(activating)]))) 
+
+tp53_alterations_score_wide
+
+```
+
+### Add annotation
+
+As discussed above we want to annotate TP53 mutants that are putative loss-of-function OR  gain-of-function.
+
+```{r}
+
+tp53_alterations_score_wide <- tp53_alterations_score_wide %>%
+  mutate(
+    # add tp53_altered annotation column
+    tp53_altered = 
+      case_when(
+        # when activating == ""
+        activating == "" &
+          # check if mutated variant AA position overlaps hotspot database 
+          ( hotspot == "Yes"  |
+              # check if sample_id has SNV+CNV mutation
+              # suggesting both alleles are mutated
+              (SNV_indel_counts >= 1 & CNV_loss_counts >=1) |
+              # check if more than 1 SNV is present
+              # suggesting both alleles are mutated
+              SNV_indel_counts >= 1 |
+              # check if SNV mutant and has
+              # Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (SNV_indel_counts >= 1 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions)) |
+              # check if CNV mutant  and has
+              # Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (CNV_loss_counts >= 1 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions)) |
+              # check if tp53 inactivating score for RNA_Seq is greater that 0.5
+              # and has Li-Fraumeni syndrome suggesting
+              # germline TP53 mutant as cancer predisposition 
+              (tp53_score > 0.5 & 
+                             grepl("Li-Fraumeni syndrome",cancer_predispositions))
+          ) ~ "loss",
+        # when activating == "Yes"
+        activating == "Yes" ~ "activated",
+        # if no evidence supports TP53 inativation or activation 
+        TRUE ~ "No evidence"
```

Comment:
```suggestion
        TRUE ~ "Other"
```

Now that I think about this a bit more, I think maybe we should rename this - it is not that there is no evidence, it is that there is not high-confidence evidence to support this.